infrastructure/rdfdb/docker/scripts/modify_web_xml.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -e
+
+WAR_FILE="/usr/local/tomcat/webapps/rdf4j-server.war"
+TEMP_DIR="/tmp/rdf4j-server"
+
+# Extract WAR file
+mkdir -p "$TEMP_DIR"
+cd "$TEMP_DIR"
+unzip -q "$WAR_FILE"
+
+# Modify web.xml
+WEB_XML="$TEMP_DIR/WEB-INF/web.xml"
+# Use sed to insert the security configurations just before the closing </web-app> tag
+sed -i '/<\/web-app>/i\
+  <security-constraint>\
+        <web-resource-collection>\
+            <web-resource-name>RDF4J Server</web-resource-name>\
+            <url-pattern>/protocol</url-pattern>\
+        </web-resource-collection>\
+        <!-- No auth-constraint here, making it publicly accessible --> \
+    </security-constraint>\
+\
+    <security-constraint>\
+        <web-resource-collection>\
+            <web-resource-name>RDF4J Server</web-resource-name>\
+            <url-pattern>/*</url-pattern>\
+        </web-resource-collection>\
+        <auth-constraint>\
+            <role-name>rdf4j-user</role-name>\
+        </auth-constraint>\
+    </security-constraint>\
+\
+    <login-config>\
+        <auth-method>BASIC</auth-method>\
```

Comment:
can we not use Digest?

Reply:
The DB is internal, has no access to the public, so I think we are ok with basic.   I could create a 1 point ticket to switch it though, doesn't look that difficult.